### PR TITLE
[Tokens]: adds support for token layers

### PR DESCRIPTION
**Pull request by AlanBreck**
This is a first attempt at support "layers" in Nala. The idea is that "universal" tokens would always be included, but then layers like "browser" or "marketing" can be added to both override and extend tokens in the universal layer.

E.g.

```css
@import "@brave/leo/tokens/css/variables-universal";
@import "@brave/leo/tokens/css/variables-browser";
```

Tokens output:

```
tokens
├── android
│  └── no change
├── css
│  ├── variables-browser.css
│  ├── variables-browser.js
│  ├── variables-browser.ts
│  ├── variables-universal.css
│  ├── variables-universal.js
│  └── variables-universal.ts
├── ios-swift
│  └── no change
├── json
│  ├── styles-browser.json
│  └── styles-universal.json
├── skia
│  └── no change
└── tailwind
   ├── browser
   │  ├── index.js
   │  ├── package.json
   │  ├── plugins
   │  ├── tokens.js
   │  ├── typography.js
   │  └── variables.css
   ├── package.json
   ├── plugins
   │  └── components
   └── universal
      ├── index.js
      ├── package.json
      ├── plugins
      ├── tokens.js
      ├── typography.js
      └── variables.css

```

### CODE REVIEW

**Comment by thypon** [comment]
Security gtg

File: src/tokens/transformation/tailwind/formatCssVars.js
Comment (AlanBreck) [inline src/tokens/transformation/tailwind/formatCssVars.js:34]: Either now, or later, I believe that this needs to be refactored to better account for this type of removal. Maybe we should remove this kind of thing at the parser level?

Maybe even in Figma (if possible) since we no longer really need to distinguish different platforms (or I'm calling them layers to disambiguate from StyleDictionary's use of "platforms").

File: src/tokens/browserFontTokens.stories.svelte
Comment (AlanBreck) [inline src/tokens/browserFontTokens.stories.svelte:1]: I'd love a programmatic way of generating well organized stories for our tokens, but as of right now, I'm not sure of the best way to do so.

Would like to minimize the amount of manual updates necessary when a new "layer" (browser, marketing, etc.) is added.

File: .github/workflows/deploy.yml
Comment (AlanBreck) [reply .github/workflows/deploy.yml:1]: I was afraid of that.

File: .storybook/global.svelte
Comment (AlanBreck) [reply .storybook/global.svelte:6]: I thought of that as well, but I think I personally lean towards keeping it explicit.

File: .storybook/global.svelte
Comment (AlanBreck) [reply .storybook/global.svelte:7]: I kind of like it as well. I will say, though, that some importers (I think SASS?) requires the `.css` to be dropped, and `@brave-leo/css/variables.browser` maybe looks a little odd?

Ultimately, I'm not super opinionated on it.

File: src/tokens/transformation/tailwind/convertCssToJs.js
Comment (AlanBreck) [reply src/tokens/transformation/tailwind/convertCssToJs.js:6]: Not for Tailwind, since the different layers build to distinct folders.

```
tailwind
├── browser
│  ├── index.js
│  ├── package.json
│  ├── plugins
│  ├── tokens.js
│  ├── typography.js
│  └── variables.css
├── package.json
├── plugins
│  └── components
└── universal
   ├── index.js
   ├── package.json
   ├── plugins
   ├── tokens.js
   ├── typography.js
   └── variables.css
```

File: src/tokens/transformation/tailwind/twFilterFonts.js
Comment (AlanBreck) [reply src/tokens/transformation/tailwind/twFilterFonts.js:6]: We could. I thought we'd leave it for the possibility of needing to exclude other path parts in future. Maybe it's best to remove, given that we'll hopefully refactor the whole thing.

File: src/tokens/transformTokens.js
Comment (AlanBreck) [reply src/tokens/transformTokens.js:35]: Not following here. Can you explain?

File: src/tokens/transformation/tailwind/twFilterFonts.js
Comment (fallaciousreasoning) [reply src/tokens/transformation/tailwind/twFilterFonts.js:6]: I think lets remove it. If we need to do something similar it shouldn't take us too long to come back up with this (hopefully)

File: src/tokens/transformTokens.js
Comment (fallaciousreasoning) [reply src/tokens/transformTokens.js:35]: Ah, I meant do we still build the android/ski stuff.

Comment (AlanBreck) [reply src/tokens/transformTokens.js:35]: Yep. This PR leaves Android, Skia, and iOS untouched.

https://github.com/brave/leo/blob/c9b960b4b76a093a7bf6e334c149da63645b9669/src/tokens/transformTokens.js#L27-L32

File: src/tokens/transformation/web/formatCss.js
Comment (AlanBreck) [reply src/tokens/transformation/web/formatCss.js]: Abstracted into a function.

https://github.com/brave/leo/blob/91f52f9a1ef5585364dd8e067f703b74a3e44eb6/src/tokens/utils.js#L73-L81

File: src/tokens/transformation/tailwind/formatFonts.js
Comment (AlanBreck) [reply src/tokens/transformation/tailwind/formatFonts.js:18]: Abstracted into a function.

https://github.com/brave/leo/blob/91f52f9a1ef5585364dd8e067f703b74a3e44eb6/src/tokens/utils.js#L73-L81

File: .storybook/global.svelte
Comment (AlanBreck) [reply .storybook/global.svelte:6]: Preface, I'm quite happy to go either way on this.

However, in favor of being explicit:
1. This change is big enough that failing loudly may be useful to alert consumers of the change.
2. The name `variables-universal.css` helps to further indicate that there's potentially more to the story than simply `variables.css`, and hints at the fact that it is itself merely one of several layers.
3. The fix for this change is pretty minimal, as it should only require an update in one file for many projects, and only a handful for more complex projects.

File: src/tokens/transformation/tailwind/twFilterFonts.js
Comment (AlanBreck) [reply src/tokens/transformation/tailwind/twFilterFonts.js]: Removed.

File: .github/workflows/build.yml
Comment (AlanBreck) [reply .github/workflows/build.yml:52]: This will get removed at the last minute. It's in their for now so as to allow the builds to pass.

File: .storybook/global.svelte
Comment (fallaciousreasoning) [reply .storybook/global.svelte:6]: Guess I'm willing to be convinced :laughing: 

Last counterpoint - for a lot of things just the base layer will be enough - I think (at the moment) brave-core only really used universal tokens.

But happy to land as is.

File: .storybook/global.svelte
Comment (fallaciousreasoning) [reply .storybook/global.svelte:7]: Oh dear. Well. Guess this is fine, just a little ugly :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
Okay, this LGTM now. Do you mind holding off on landing it until @nullhook's PR lands - I need to get that into brave-core, and it'd be nice not to have to make this change at the same time.

**Comment by AlanBreck** [comment]
> Okay, this LGTM now. Do you mind holding off on landing it until @nullhook's PR lands - I need to get that into brave-core, and it'd be nice not to have to make this change at the same time.

No problem. I thought we'd wait on @petemill to weigh in on the two open comments. Good points raised.

**Comment by petemill** [comment]
I agree that variables.css _could_ seem like it's a superset of everything else, but in practice when I tried it out, it didn't. It could also be interpreted as "base" (i.e. an implied variables-base.css though that doesn't match up with figma).

In both questions, I'm ok with whatever is preferred here, but if you need a third decision then I'd vote for:
- variables-browser.css instead of variables.browser.css, because as far as I can tell we usually reset .xyz.filetype for when xyz is still providing meaning the _type_ of file, e.g. button.stories.svelte or variables.css.ts.
- variables.css instead of variables-universal.css as a compromise because it will be used the most and it seems both of you don't like the dash-case from above (would you prefer another casing scheme?).

**Comment by fallaciousreasoning** [comment]
Sweet, I'm on board with @petemill's tie-breaking vote :laughing: `variables.css` for universal styles and `-<additional_overrides>.css` for the extra styles.

**Comment by AlanBreck** [comment]
> In both questions, I'm ok with whatever is preferred here, but if you need a third decision then I'd vote for:
> 
> * variables-browser.css instead of variables.browser.css, because as far as I can tell we usually reset .xyz.filetype for when xyz is still providing meaning the _type_ of file, e.g. button.stories.svelte or variables.css.ts.
> * variables.css instead of variables-universal.css as a compromise because it will be used the most and it seems both of you don't like the dash-case from above (would you prefer another casing scheme?).
> 
> I agree that variables.css _could_ seem like it's a superset of everything else, but in practice when I tried it out, it didn't. It could also be interpreted as "base" (i.e. an implied variables-base.css though that doesn't match up with figma).

Updated as per these recommendations (only change being removing `-universal`).

**Review by fallaciousreasoning — APPROVED** [review]

**Review by fallaciousreasoning — APPROVED** [review]

**Pull request merged**
